### ISSUE TITLE
Questions regarding the reported performance and evaluation details

### ISSUE THREAD

**Issue by wjdgns0914** [mli0603/openpi-comet#18]
I am writing this to ask a few questions regarding the performance reported in your report and the GitHub repo.

1. Clarification on Validation Data
To ensure I have the correct understanding, does the "validation performance" mentioned in the report refer to the performance on the first 10 instances among the 20 instances specified in metadata/test_instances.csv from the following dataset? (Link: https://huggingface.co/datasets/behavior-1k/2025-challenge-task-instances)

2. Discrepancy in Performance Metrics
  <img width="1667" height="490" alt="Image" src="https://github.com/user-attachments/assets/0109dce1-066a-406a-971b-ff8a4d21f22b" />
  
In Table 1, the "Full Task success rate" on validation data is reported as 14.4%, and the "Q-score" is 34.5%. However, calculating the average performance based on the 25 tasks shown in Figure 4 results in (10+10+10+10+10+10+10+10+8+8+6+5+4+4+3+3+2+2+2+2+1+1+1+1+1)/500 = 0.268 (26.8%). Could you please clarify what this 26.8% figure represents?

3. Evaluation Methodology for Multiple Models
On GitHub, it is mentioned that a 34.5% Q-score was achieved using only two pretrained models. How was this performance calculated across the two models (pi05-b1kpt12-cs32 and pi05-b1kpt50-cs32)? 
Specifically, was the score obtained by:
- Taking the higher performance between the two checkpoints for each task?
- Or, was an instance counted as a success if either of the two policies succeeded? 
For example, if for Task 0, pi05-b1kpt12-cs32 succeeds on instances 1–3 and pi05-b1kpt50-cs32 succeeds on instances 4–10, would the success rate for that task be 60% (the higher individual rate) or 100% (the combined success)?

4. Reproducibility and Alignment
If I follow the instructions provided in the README using the two released checkpoints, should the Q-score be fully reproducible? I have observed slight differences between the reported performance and my own results. I am wondering if there are any specific details or environment settings not covered in the README that I should be aware of to align the results.

Thank you for your time and for sharing your valuable work. I look forward to your response.
@DelinQu

**Comment by DelinQu** [comment]
Hi [wjdgns0914](https://github.com/wjdgns0914),

Regarding the discrepancy you observed, I believe there may be a misunderstanding around the Accuracy (ACC) reported in Figure 4, and the numbers were not updated to the latest statistics. The ACC shown there does not correspond to the task success of the models we released. Instead, it reflects the aggregated success rate of all post-training inner models (denoted as “#Per-instance Ensembles”) collected during our Rejection Sampling Fine-Tuning (RFT / Post-training++) process. These statistics are computed over internally generated rollout data used for post-training, rather than the final released evaluation models.

1. For clarity, the metrics reported in Figure 4 (#Per-instance Ensembles) are:

Aggregated validation Q-score: 0.611 (BEHAVIOR-1K validation set)
Validation success rate: 35.0% across 50 tasks under the standard evaluation protocol

2. In contrast, the performance of our released two models is:

Aggregated validation Q-score: 0.345 (BEHAVIOR-1K validation set)
Validation success rate: 15.8% across 50 tasks

While more extensive model combinations or ensembling across additional inner models could potentially yield higher scores, we intentionally limited the public release to two models in order to preserve system simplicity, clarity, and reproducibility.

I will update the report shortly to make this distinction explicit and avoid potential misunderstanding. 

## Post-challenge result of Released Two Models 

q score 0.345

Final Q Score: 0.34531561216561196
Final Time Score: 0.5494098277653365
Final Base Distance Score: 2.965259999646887
Final Left Distance Score: 0.2603500379877089
Final Right Distance Score: 0.26781900022218713

ACC 79 /  500

Task Name                                          QScore==1 Count
-----------------------------------------------------------------
turning_on_radio                                   7              
picking_up_trash                                   8              
putting_away_Halloween_decorations                 2              
cleaning_up_plates_and_food                        1              
can_meat                                           0              
setting_mousetraps                                 0              
hiding_Easter_eggs                                 2              
picking_up_toys                                    0              
rearranging_kitchen_furniture                      2              
putting_up_Christmas_decorations_inside            0              
set_up_a_coffee_station_in_your_kitchen            0              
putting_dishes_away_after_cleaning                 3              
preparing_lunch_box                                1              
loading_the_car                                    1              
carrying_in_groceries                              0              
bringing_in_wood                                   3              
moving_boxes_to_storage                            6              
bringing_water                                     9              
tidying_bedroom                                    0              
outfit_a_basic_toolbox                             1              
sorting_vegetables                                 1              
collecting_childrens_toys                          0              
putting_shoes_on_rack                              1              
boxing_books_up_for_storage                        0              
storing_food                                       0              
clearing_food_from_table_into_fridge               0              
assembling_gift_baskets                            0              
sorting_household_items                            0              
getting_organized_for_work                         0              
clean_up_your_desk                                 0              
setting_the_fire                                   0              
clean_boxing_gloves                                1              
wash_a_baseball_cap                                7              
wash_dog_toys                                      0              
hanging_pictures                                   0              
attach_a_camera_to_a_tripod                        0              
clean_a_patio                                      0              
clean_a_trumpet                                    0              
spraying_for_bugs                                  0              
spraying_fruit_trees                               2              
make_microwave_popcorn                             7              
cook_cabbage                                       0              
chop_an_onion                                      0              
slicing_vegetables                                 2              
chopping_wood                                      0              
cook_hot_dogs                                      7              
cook_bacon                                         1              
freeze_pies                                        0              
canning_food                                       0              
make_pizza                                         0              


Total QScore==1: 75 files



### Per-instance Ensembles across ALL Inner Models 


Q score 0.611

"q_score": 0.6110833444333443,
"time_score": 0.5767210448030198,
"base_distance_score": 2.0014505700858076,
"left_distance_score": 0.2553154712504806,
"right_distance_score": 0.26345036466049093

ACC 175 /  500

Task Name                                          QScore==1 Count
-----------------------------------------------------------------
turning_on_radio                                   10             
picking_up_trash                                   10             
putting_away_Halloween_decorations                 7              
cleaning_up_plates_and_food                        2              
can_meat                                           0              
setting_mousetraps                                 5              
hiding_Easter_eggs                                 7              
picking_up_toys                                    0              
rearranging_kitchen_furniture                      4              
putting_up_Christmas_decorations_inside            0              
set_up_a_coffee_station_in_your_kitchen            4              
putting_dishes_away_after_cleaning                 6              
preparing_lunch_box                                5              
loading_the_car                                    5              
carrying_in_groceries                              3              
bringing_in_wood                                   7              
moving_boxes_to_storage                            10             
bringing_water                                     10             
tidying_bedroom                                    0              
outfit_a_basic_toolbox                             3              
sorting_vegetables                                 2              
collecting_childrens_toys                          2              
putting_shoes_on_rack                              7              
boxing_books_up_for_storage                        0              
storing_food                                       0              
clearing_food_from_table_into_fridge               3              
assembling_gift_baskets                            0              
sorting_household_items                            0              
getting_organized_for_work                         0              
clean_up_your_desk                                 0              
setting_the_fire                                   0              
clean_boxing_gloves                                2              
wash_a_baseball_cap                                10             
wash_dog_toys                                      1              
hanging_pictures                                   4              
attach_a_camera_to_a_tripod                        1              
clean_a_patio                                      0              
clean_a_trumpet                                    0              
spraying_for_bugs                                  3              
spraying_fruit_trees                               4              
make_microwave_popcorn                             10             
cook_cabbage                                       4              
chop_an_onion                                      3              
slicing_vegetables                                 9              
chopping_wood                                      0              
cook_hot_dogs                                      10             
cook_bacon                                         1              
freeze_pies                                        1              
canning_food                                       0              
make_pizza                                         0              


Total QScore==1: 175 files

**Comment by DelinQu** [comment]
1. Yes. The “validation performance” reported in the report refers to evaluation on the **first 10 instances** (out of the 20 listed in `metadata/test_instances.csv`) for each task in the BEHAVIOR-1K validation split, following the official challenge evaluation protocol.

2. There may be a misunderstanding around the Accuracy (ACC) reported in Figure 4, and the numbers were not updated to the latest statistics. See the above reply.

3. The reported Q-score using two pretrained models is computed via per-task aggregation, not by ensembling or merging instance results. In your example, the task success rate would be 70%. We may evaluate the same model multiple times per task, as BEHAVIOR rollouts exhibit noticeable stochasticity. For example, on the turning_on_radio task, we observed success rate variations of approximately 40%–80% across repeated runs.

4. Yes, following the README with the two released checkpoints should reproduce the reported Q-score up to minor variance. Small discrepancies can arise from:
* Environment nondeterminism (physics, rendering), use the latest BEHAVIOR-1K v3.7.2
* Normalization statistics inheritance
* Wrapper versions or rollout seeds
These differences are typically small and should not materially affect the aggregated score.

Hope this clarifies things.

**Comment by wjdgns0914** [comment]
> 1. Yes. The “validation performance” reported in the report refers to evaluation on the **first 10 instances** (out of the 20 listed in `metadata/test_instances.csv`) for each task in the BEHAVIOR-1K validation split, following the official challenge evaluation protocol.
> 2. There may be a misunderstanding around the Accuracy (ACC) reported in Figure 4, and the numbers were not updated to the latest statistics. See the above reply.
> 3. The reported Q-score using two pretrained models is computed via per-task aggregation, not by ensembling or merging instance results. In your example, the task success rate would be 70%. We may evaluate the same model multiple times per task, as BEHAVIOR rollouts exhibit noticeable stochasticity. For example, on the turning_on_radio task, we observed success rate variations of approximately 40%–80% across repeated runs.
> 4. Yes, following the README with the two released checkpoints should reproduce the reported Q-score up to minor variance. Small discrepancies can arise from:
> 
> * Environment nondeterminism (physics, rendering), use the latest BEHAVIOR-1K v3.7.2
> * Normalization statistics inheritance
> * Wrapper versions or rollout seeds
>   These differences are typically small and should not materially affect the aggregated score.
> 
> Hope this clarifies things.

Thank you so much for the detailed response. It cleared up a lot of my doubts. 
Could I ask two more questions?

1. Actually, I also evaluated the two public checkpoints, and the task-by-task ensemble performance resulted in a q-score of about 26%. However, hearing your explanation, I realized that what I obtained was the performance from evaluating each task exactly once (for reference, I got 50% on the turning_on_radio task using the pt12 checkpoint). Since you mentioned to 'evaluate the same model multiple times per task' and stated that the performance for turning_on_radio ranges from 40% to 80%, it seems certain that the final q-score would be significantly higher if I test multiple times per task and take the highest performance. What do you think about this assumption?

2. Do I need to use a different `norm_stats.json` for each task? When testing multiple tasks with the public checkpoints, I'm wondering if it's okay to consistently use the `norm_stats.json` calculated from **the entire task data** with 'pi05_b1k-base'.

Thanks again, and I look forward to hearing from you. @DelinQu

**Comment by DelinQu** [comment]
(1) Yes, running the same task multiple times can yield a wide range of outcomes due to evaluation variance.
The reported numbers came from all historical per-task evaluations during the rollout collection (across the two models). I also re-ran a clean single-pass evaluation afterward, which gave a score of ~30%. So I’m not sure why your single-run result shows a much larger gap, but it may come from simulator variance or setup differences. A more objective protocol would indeed be similar to LIBERO — running each task 50 times with different seeds and reporting the mean. However, for BEHAVIOR-1K, this quickly becomes prohibitively expensive: evaluating 50 tasks × 10 instances already has a very high simulation cost :(

(2) No, you do not need a different norm_stats.json per task. During evaluation, the model automatically inherits the normalization statistics stored in the checkpoint. Per-task normalization is only meaningful if the model was explicitly fine-tuned with per-task stats. Otherwise, using per-task norm_stats during evaluation is usually harmful rather than helpful.

**Comment by wjdgns0914** [comment]
I really appreciate the detailed response. It helped a lot!!
